services/oauth/src/main/java/org/mule/services/oauth/internal/AbstractOAuthDancer.java

```diff
@@ -238,7 +223,7 @@ protected TokenResponse invokeTokenUrl(String tokenUrl, Map<String, String> toke
    * @param resourceOwnerId id of the user.
    * @return oauth state
    */
-  protected DefaultResourceOwnerOAuthContext getContextForResourceOwner(final String resourceOwnerId) {
+  public ResourceOwnerOAuthContext getContextForResourceOwner(final String resourceOwnerId) {
```

Comment:
client credentials reuses this, putting its only context witha constant as a key, here: https://github.com/mulesoft/mule/pull/4487/files#diff-88a5f3c13546df357111856a8829052cR138

When using just the api interfaces, this is not exposef for client credentials